core/evm.go

```diff
@@ -48,6 +51,32 @@ type ChainContext interface {
 
 // NewEVMBlockContext creates a new context for use in the EVM.
 func NewEVMBlockContext(header *types.Header, chain ChainContext, author *common.Address) vm.BlockContext {
+	var (
+		predicateResults *results.PredicateResults
+		err              error
+	)
+	if len(header.Extra) > params.DynamicFeeExtraDataSize {
+		// Prior to the DUpgrade, the VM enforces the extra data is smaller than or equal to this size.
+		// After the DUpgrade, the VM pre-verifies the extra data past the dynamic fee rollup window is
+		// valid.
```

Comment:
We could, it would just require changing the `ChainContext` here to also return the `ChainConfig`, which requires changing some further tests to make sure that `ChainContext` / `ChainConfig` are always populated where some tests do not currently ensure that.

Specifically, this introduces the new requirement that the `ChainContext` actually needs to be non-nil. Even though this seems like a reasonable assumption, `addTx` within `core/chain_makers.go` currently passes in `nil` instead of an actual blockchain instance, so this causes the existing tests using it to panic.